### DISCUSSION TITLE
password reset problem

### DISCUSSION THREAD

**Discussion by BeckeBauer** [kimai/kimai#3295] [General]
I recently changed from php74 to php80. Consequence is that passwords are not longer working.

When I try to reset password by e-mail, I get the following error:

`[2022-05-08 10:28:18] request.INFO: Matched route "fos_user_resetting_send_email". {"route":"fos_user_resetting_send_email","route_parameters":{"_route":"fos_user_resetting_send_email","_controller":"App\\Controller\\Security\\PasswordResetController::sendEmailAction","_locale":"de"},"request_uri":"https://mydomain/de/resetting/send-email","method":"POST"} []
[2022-05-08 10:28:18] security.INFO: Populated the TokenStorage with an anonymous Token. [] []
[2022-05-08 10:28:19] php.CRITICAL: Uncaught Error: Undefined constant "XSL_SECPREF_READ_FILE" {"exception":"[object] (Error(code: 0): Undefined constant \"XSL_SECPREF_READ_FILE\" at /kimai2/vendor/lorenzo/pinky/src/pinky.php:21)"} []
[2022-05-08 10:28:19] request.CRITICAL: Uncaught PHP Exception Error: "Undefined constant "XSL_SECPREF_READ_FILE"" at /kimai2/vendor/lorenzo/pinky/src/pinky.php line 21 {"exception":"[object] (Error(code: 0): Undefined constant \"XSL_SECPREF_READ_FILE\" at /kimai2/vendor/lorenzo/pinky/src/pinky.php:21)"} []` 

any help is appreciated

**Comment by kevinpapst** [comment]
That is and official XSL constant: https://www.php.net/manual/en/xsl.constants.php
And Kimai requires the XSL extension to be installed: https://github.com/kevinpapst/kimai2/blob/master/composer.json#L19

There are two ways I see to get this error:

- you install composer dependencies with `--ignore-platform-reqs`
- you did not execute composer after upgrading PHP (which is definitely necessary)

**Comment by BeckeBauer** [comment]
That was it! I activated the php XSL extension and got my next error. Then, I simply activated all available php extension and could login again (don't know which extension(s) were missing as well).

Thanks for always helping 👍

**Comment by BeckeBauer** [comment]
Maybe to avoid such a problem for users like me which do not check the composer.json but use the "Kimai Doctor Tool": The latter could be amended and instead of (or in addition to) the XML extension as being required, the XSL extension could be mentioned here:
![Webaufnahme](https://user-images.githubusercontent.com/51003649/167312479-18863dd9-0bc9-4bb9-b761-de9449fbc464.jpeg)

**Reply by kevinpapst** [reply]
You shouldn't be able to install Kimai without those extensions. 
The doctor is more of a leftover from the times where I supported FTP installations. 
How often does someone upgrade a PHP version? 😁

**Comment by BeckeBauer** [comment]
I just upgraded php74 to php80 (and I though to activate only the necessary php extensions and not all extensions available as done previously) so there was no installing Kimai - would you recommend to do so after "php upgrade"?
